src/cli/cli.cc

```diff
@@ -1229,37 +1240,171 @@ bool isSetupComplete (SSC::String platform) {
   return funcs[platform]();
 }
 
-int main (const int argc, const char* argv[]) {
-  defaultTemplateAttrs = {{ "ssc_version", SSC::VERSION_FULL_STRING }};
-  if (argc < 2) {
-    printHelp("ssc");
+void run (const String& targetPlatform, Map& settings, const Paths& paths, const bool& flagDebugMode, const bool& flagHeadless, const String& argvForward, AndroidCliState& androidState) {
+  if (targetPlatform == "ios-simulator") {
+    String app = (settings["build_name"] + ".app");
+    auto pathToApp = paths.platformSpecificOutputPath / app;
+    runIOSSimulator(pathToApp, settings);
+  } else if (targetPlatform == "android" || targetPlatform == "android-emulator") {
+    if (!getAdbPath(androidState)) {
+      exit(1);
+    }
+
+    if (targetPlatform == "android-emulator" && !androidState.emulatorRunning) {
+      if (!initAndStartAndroidEmulator(androidState)) {
+        exit(1);
+      }
+    }
+
+    if (!initAndStartAndroidApp(androidState, flagDebugMode)) {
+      exit(1);
+    }
+
     exit(0);
+  } else {
+    auto executable = Path(settings["build_name"] + (platform.win ? ".exe" : ""));
+    auto exitCode = runApp(paths.pathBin / executable, argvForward, flagHeadless);
+    return exit(exitCode);
   }
 
-  auto is = [](const String& s, const auto& p) -> bool {
-    return s.compare(p) == 0;
-  };
+  // Fixes "subcommand 'ssc run' is not supported.""
+  log("App failed to run, please contact support.");
+  exit(1);
+}
 
-  auto optionValue = [](
-    const String& c, // command
-    const String& s,
-    const String& p
-  ) -> String {
-    auto string = p + String("=");
-    if (String(s).find(string) == 0) {
-      auto value = s.substr(string.size());
-      if (value.size() == 0) {
-        value = rc[c + "_" + p];
+void handleArgument(
+  Map& argumentsWithValue,
+  std::unordered_set<String>& argumentsWithoutValue,
+  const String& key,
+  const String& value,
+  const String& subcommand
+) {
+  if (argumentsWithValue.count(key) > 0 || argumentsWithoutValue.find(key) != argumentsWithoutValue.end()) {
+    std::cerr << "ERROR: Argument '" << key << "' is used more than once." << std::endl;
```

Comment:
It is valid for some arguments with to repeat, eg specifiying multiple `--env=VARIABLE`
Do the changes cover this use case?

Reply:
`--env` is a special case in the `handleArgument`

Reply:
I want to add that it was a special case before as well, but yes, we can make this a general case later